pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -413,17 +426,91 @@ func setNodeScoresToZero(nodes []v1.Node) (priorityList schedulerapi.HostPriorit
 	return
 }
 
+func getRequestedVolumeSet(ns string, requestedVolumes []v1.Volume) map[string]struct{} {
+	// create a set of all volumes needed
+	volumeSet := map[string]struct{}{}
+	for _, volume := range requestedVolumes {
+		if volume.PersistentVolumeClaim != nil {
+			if v, ok := pvcToPvMapCache.Load(getQualifiedName(ns, volume.PersistentVolumeClaim.ClaimName)); ok {
+				entry := v.(*pvcToPVEntry)
+				if !strings.EqualFold(entry.AccessMode, string(v1.ReadWriteMany)) &&
+					!strings.EqualFold(entry.AccessMode, string(v1.ReadOnlyMany)) {
+					if v, ok := pvToVolumeMapCache.Load(entry.VolumeName); ok {
+						diskName := v.(string)
+						volumeSet[diskName] = struct{}{}
+					}
+				}
+			}
+		}
+	}
+	return volumeSet
+}
+
 func onPvcUpdate(obj interface{}, newObj interface{}) {
 	// Cast the obj to pvc
 	pvc := newObj.(*v1.PersistentVolumeClaim)
-	klog.Infof("Updating pvcToPv map cache.PVC: %s PV: %s, AccessMode: %v", pvc.Name, pvc.Spec.VolumeName, pvc.Spec.AccessModes)
-	cacheValue := cacheEntry{VolumeName: pvc.Spec.VolumeName, AccessMode: fmt.Sprintf("%v", pvc.Spec.AccessModes[0])}
-	pvcToPvMapCache.AddOrUpdate(pvc.Name, &cacheValue)
+	addPvcEntry(pvc)
+}
+
+func addPvcEntry(pvc *v1.PersistentVolumeClaim) {
+	qualifiedName := getQualifiedName(pvc.Namespace, pvc.Name)
+	klog.Infof("Updating pvcToPv map cache.PVC: %s PV: %s, AccessMode: %v", qualifiedName, pvc.Spec.VolumeName, pvc.Spec.AccessModes)
+	cacheValue := pvcToPVEntry{VolumeName: pvc.Spec.VolumeName, AccessMode: fmt.Sprintf("%v", pvc.Spec.AccessModes[0])}
+	pvcToPvMapCache.Store(qualifiedName, &cacheValue)
 }
 
 func onPvcDelete(obj interface{}) {
 	// Cast the obj to pvc
 	pvc := obj.(*v1.PersistentVolumeClaim)
+	qualifiedName := getQualifiedName(pvc.Namespace, pvc.Name)
 	klog.Infof("Deleting pvcToPv map cache entry. PVC: %s PV: %s", pvc.Name, pvc.Spec.VolumeName)
-	pvcToPvMapCache.Delete(pvc.Name)
+	pvcToPvMapCache.Delete(qualifiedName)
+}
+
+func onPVUpdate(obj interface{}, newObj interface{}) {
```

Comment:
np: onPvUpdate, addPvEntry, onPvDelete or change the respective pvc function names to uppercase but I think the lowercase is the more common convention.